### PR TITLE
Update project amend date

### PR DESCRIPTION
**Pull request by wheelsandcogs**

### CODE REVIEW

File: lib/resolvers/project.js [hunk 1]
```diff
@@ -83,6 +90,12 @@ module.exports = ({ models }) => async ({ action, data, id, meta }, transaction)
       licenceNumber: project.licenceNumber || licenceNumber
     };
 
+    const lastAmendedDate = await getLastAmendedDate();
```
Comment (lennym) [inline lib/resolvers/project.js]: I don't think this is needed. The date either doesn't need to be set (i.e. it's a first time application and so `grantedVersion` won't exist) or it's "now".

File: lib/resolvers/project.js [hunk 2]
```diff
@@ -83,6 +90,12 @@ module.exports = ({ models }) => async ({ action, data, id, meta }, transaction)
...
+
+    if (lastAmendedDate) {
```
Comment (lennym) [inline lib/resolvers/project.js]:
```suggestion
    if (grantedVersion) {
```

File: lib/resolvers/project.js [hunk 3]
```diff
@@ -83,6 +90,12 @@ module.exports = ({ models }) => async ({ action, data, id, meta }, transaction)
...
+      patch.amendedDate = lastAmendedDate;
```
Comment (lennym) [inline lib/resolvers/project.js]:
```suggestion
      patch.amendedDate = moment().toISOString();
```

File: test/resolvers/project.js [hunk 4]
```diff
@@ -9,6 +9,10 @@ const projectId2 = 'd01588c4-cdca-461f-95de-f2bc2b95c9b0';
 
 const establishmentId = 8201;
 
+const isNowish = (date) => {
+  return moment(date).isBetween(moment().subtract(5, 'seconds'), moment().add(5, 'seconds'));
+};
```
Comment (joefitter) [inline test/resolvers/project.js]: is this now a thing in 2 places?

File: lib/resolvers/project.js [hunk 5]
```diff
@@ -25,6 +25,13 @@ module.exports = ({ models }) => async ({ action, data, id, meta }, transaction)
       .first();
   };
 
+  const getLastAmendedDate = () => {
+    return ProjectVersion.query(transaction)
+      .where({ projectId: id, status: 'granted' })
+      .orderBy('updatedAt', 'desc')
+      .then(versions => versions.length > 1 ? versions[0].updatedAt : null);
+  };
```
Comment (lennym) [inline lib/resolvers/project.js]: This function can be removed per my comments below.

File: test/resolvers/project.js (hunk 4 above)
Comment (lennym) [reply test/resolvers/project.js]: Yes, but also meh.

**Pull request merged**